Traceback (most recent call last):
  File "C:\Users\Bruna\PycharmProjects\PrevisorReconhecimento\.venv\Lib\site-packages\jupyter_cache\executors\utils.py", line 58, in single_nb_execution
    executenb(
  File "C:\Users\Bruna\PycharmProjects\PrevisorReconhecimento\.venv\Lib\site-packages\nbclient\client.py", line 1314, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Bruna\PycharmProjects\PrevisorReconhecimento\.venv\Lib\site-packages\jupyter_core\utils\__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Bruna\AppData\Local\Programs\Python\Python312\Lib\asyncio\base_events.py", line 687, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "C:\Users\Bruna\PycharmProjects\PrevisorReconhecimento\.venv\Lib\site-packages\nbclient\client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "C:\Users\Bruna\PycharmProjects\PrevisorReconhecimento\.venv\Lib\site-packages\nbclient\client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "C:\Users\Bruna\PycharmProjects\PrevisorReconhecimento\.venv\Lib\site-packages\nbclient\client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
#Supress warnings
if not sys.warnoptions:
    warnings.simplefilter("ignore")
    os.environ["PYTHONWARNINGS"] = "ignore" # Also affect subprocesses
    
# Criar pipeline com normalizaÃ§Ã£o e modelo de RegressÃ£o LogÃ­stica
pipeline_2010_2017_modelo2 = Pipeline([
    ('scaler', StandardScaler()),
    ('logreg', LogisticRegression(random_state=1, solver='lbfgs'))
])

# Definir hiperparÃ¢metros para o Grid Search
param_grid_2010_2017_modelo2 = {
    'logreg__C': [0.01, 0.1, 1, 10],  # RegularizaÃ§Ã£o
    'logreg__max_iter': [1000],  # NÃºmero mÃ¡ximo de iteraÃ§Ãµes
    'logreg__fit_intercept': [True, False],  # Se deve ajustar o intercepto
    'logreg__tol': [1e-4, 1e-3],  # TolerÃ¢ncia para convergÃªncia
    'logreg__penalty': ['l1', 'l2'],  # Tipos de penalidade
    'logreg__solver': ['saga'],
}

# Grid Search com validaÃ§Ã£o cruzada
grid_search_2010_2017_modelo2 = GridSearchCV(pipeline_2010_2017_modelo2, param_grid_2010_2017_modelo2, cv=2, scoring='f1_weighted', n_jobs=-1)

grid_search_2010_2017_modelo2.fit(X_train_2010_2017_modelo2, y_train_2010_2017_modelo2)

# Melhor modelo encontrado
logreg_best_2010_2017_modelo2 = grid_search_2010_2017_modelo2.best_estimator_

print(f"Melhores hiperparÃ¢metros: {grid_search_2010_2017_modelo2.best_params_}")

# F1 score no conjunto de teste
y_pred_2010_2017_modelo2 = logreg_best_2010_2017_modelo2.predict(X_test_2010_2017_modelo2)

f1_2010_2017_modelo2 = f1_score(y_test_2010_2017_modelo2, y_pred_2010_2017_modelo2, average='weighted')
print(f"F1 Score: {f1_2010_2017_modelo2:.6f}")

classification_report_2010_2017_modelo2 = metrics.classification_report(y_test_2010_2017_modelo2, y_pred_2010_2017_modelo2, digits=6, target_names=['Reconhecido', 'NÃ£o reconhecido'])
print('Classification report para modelo 2 (2010 - 2017): \n')
print(classification_report_2010_2017_modelo2)
------------------


[1;31m---------------------------------------------------------------------------[0m
[1;31mNameError[0m                                 Traceback (most recent call last)
Cell [1;32mIn[4], line 2[0m
[0;32m      1[0m [38;5;66;03m#Supress warnings[39;00m
[1;32m----> 2[0m [38;5;28;01mif[39;00m [38;5;129;01mnot[39;00m [43msys[49m[38;5;241m.[39mwarnoptions:
[0;32m      3[0m     warnings[38;5;241m.[39msimplefilter([38;5;124m"[39m[38;5;124mignore[39m[38;5;124m"[39m)
[0;32m      4[0m     os[38;5;241m.[39menviron[[38;5;124m"[39m[38;5;124mPYTHONWARNINGS[39m[38;5;124m"[39m] [38;5;241m=[39m [38;5;124m"[39m[38;5;124mignore[39m[38;5;124m"[39m [38;5;66;03m# Also affect subprocesses[39;00m

[1;31mNameError[0m: name 'sys' is not defined

